JetStreamDriver.js

```diff
@@ -508,7 +528,28 @@ class Driver {
         }
     }
 
-    resultsObject()
+    resultsObject(newStyle = false) {
+        if (newStyle)
+            return this.resultsObjectNew();
+        return this.resultsObjectOld();
+    }
+
+    resultsObjectNew() {
+        const results = {__proto__: null};
+        for (const benchmark of this.benchmarks) {
+            if (!benchmark.isDone)
+                continue;
```

Comment:
IIUC, the only time `isDone` is false is when the benchmark failed right? Maybe we should add `results[benchmark.name] = "FAILED"` to the results object if so? Although, maybe we just abort out when a benchmark fails so this might not even be reachable?

Reply:
`isDone` is also false for the ones that haven't run yet, this is how I currently get the incremental updates out

- I've added an existing `_state` instance variable to make this more explicit.
- Storting "FAILED" when there are benchmark errors (top-level / setup errors are still captured separately)